### PR TITLE
Make the auth device wizard look pretty

### PR DESCRIPTION
**Pull request by bl-nero**
Contributes to https://github.com/gravitational/teleport/issues/36232
Contributes to https://github.com/gravitational/teleport/issues/37616

This PR depends on PR https://github.com/gravitational/teleport/pull/38260.

This UI is hidden until ready; to view it, set `grv_new_add_auth_device_dialog` to `true` in the app's local storage. A follow-up PR will turn this one on and remove the legacy `AddDevice` dialog.

[Figma designs](https://www.figma.com/file/8DQZl1mYUPLgvKgmA6zPOM/In-Progress-Work?type=design&node-id=7052-56894&mode=design&t=DxwciTM9fGYvG46J-0)

Screenshot 2024-02-16 at 21 40 19
Screenshot 2024-02-16 at 21 40 41
Screenshot 2024-02-16 at 21 40 54
Screenshot 2024-02-16 at 21 41 18

### CODE REVIEW

**Review by zmb3 — APPROVED** [review]
Looks great!

File: web/packages/teleport/src/Account/ManageDevices/AddAuthDeviceWizard/AddAuthDeviceWizard.tsx [hunk 1]
```diff
@@ -203,8 +208,18 @@ export function ReauthenticateStep({
...
+              <Box flex="1">
+                <ButtonPrimary block={true} type="submit">
+                  Verify my identity
+                </ButtonPrimary>
+              </Box>
+              <Box flex="1">
+                <ButtonSecondary block={true} onClick={onClose}>
+                  Cancel
+                </ButtonSecondary>
+              </Box>
```
Comment (gzdunek) [inline web/packages/teleport/src/Account/ManageDevices/AddAuthDeviceWizard/AddAuthDeviceWizard.tsx:221]: You should be able to get rid of `<Box flex="1">` wrappers.

File: web/packages/teleport/src/Account/ManageDevices/AddAuthDeviceWizard/AddAuthDeviceWizard.tsx [hunk 2]
```diff
@@ -273,14 +290,26 @@ export function CreateDeviceStep({
...
+          {stepIndex === 0 ? (
+            <ButtonSecondary block={true} onClick={onClose}>
+              Cancel
+            </ButtonSecondary>
+          ) : (
+            <ButtonSecondary block={true} onClick={prev}>
+              Back
+            </ButtonSecondary>
+          )}
+        </Box>
```
Comment (gzdunek) [inline web/packages/teleport/src/Account/ManageDevices/AddAuthDeviceWizard/AddAuthDeviceWizard.tsx:311]: Same here.

File: web/packages/teleport/src/Account/ManageDevices/AddAuthDeviceWizard/AddAuthDeviceWizard.tsx [hunk 3]
```diff
@@ -475,30 +507,84 @@ export function SaveDeviceStep({
...
+                  {usage === 'passwordless'
+                    ? 'Save the Passkey'
+                    : 'Save the MFA method'}
+                </ButtonPrimary>
+              </Box>
+              <Box flex="1">
+                <ButtonSecondary block={true} onClick={prev}>
+                  Back
+                </ButtonSecondary>
+              </Box>
```
Comment (gzdunek) [inline web/packages/teleport/src/Account/ManageDevices/AddAuthDeviceWizard/AddAuthDeviceWizard.tsx:522]: Same here.

**Review by gzdunek — APPROVED** [review]

**Pull request merged**